src/PluginsSystem/Microsoft.Performance.Toolkit.Plugins.Runtime/PluginSourceRepository.cs

```diff
@@ -0,0 +1,142 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Specialized;
+using System.Linq;
+using Microsoft.Performance.SDK;
+using Microsoft.Performance.SDK.Processing;
+using Microsoft.Performance.SDK.Runtime;
+using Microsoft.Performance.Toolkit.Plugins.Core.Discovery;
+using Microsoft.Performance.Toolkit.Plugins.Runtime.Common;
+
+namespace Microsoft.Performance.Toolkit.Plugins.Runtime
+{
+    /// <summary>
+    ///    Represents a repository of <see cref="PluginSource"/>s.
+    /// </summary>
+    public sealed class PluginSourceRepository
+        : IRepository<PluginSource>
```

Comment:
this could probably be abstracted into some type of generic abstract class

Reply:
Updated